### PR TITLE
[Narwhal] Minor test improvements 

### PR DESCRIPTION
**Pull request by vvp**
This PR adds few fixes that came up during #2537 review.

### CODE REVIEW

File: node/narwhal/src/helpers/storage.rs [hunk 1]
```diff
@@ -602,22 +602,22 @@ mod tests {
...
         storage: &Storage<N>,
-        committees: Vec<(u64, Committee<N>)>,
-        rounds: Vec<(u64, IndexSet<(Field<N>, Field<N>, Address<N>)>)>,
-        certificates: Vec<(Field<N>, BatchCertificate<N>)>,
-        batch_ids: Vec<(Field<N>, u64)>,
-        transmissions: HashMap<TransmissionID<N>, (Transmission<N>, IndexSet<Field<N>>)>,
+        committees: &Vec<(u64, Committee<N>)>,
+        rounds: &Vec<(u64, IndexSet<(Field<N>, Field<N>, Address<N>)>)>,
+        certificates: &Vec<(Field<N>, BatchCertificate<N>)>,
+        batch_ids: &Vec<(Field<N>, u64)>,
```
Comment (ljedrz) [inline node/narwhal/src/helpers/storage.rs]:
```suggestion
        committees: &[(u64, Committee<N>)],
        rounds: &[(u64, IndexSet<(Field<N>, Field<N>, Address<N>)>)],
        certificates: &[(Field<N>, BatchCertificate<N>)],
        batch_ids: &[(Field<N>, u64)],
```

File: node/narwhal/src/helpers/storage.rs [hunk 2]
```diff
@@ -602,22 +602,22 @@ mod tests {
...
+        transmissions: &HashMap<TransmissionID<N>, (Transmission<N>, IndexSet<Field<N>>)>,
     ) {
         // Ensure the committees are well-formed.
-        assert_eq!(storage.committees_iter().collect::<Vec<_>>(), committees);
+        assert_eq!(storage.committees_iter().collect::<Vec<_>>(), *committees);
```
Comment (ljedrz) [inline node/narwhal/src/helpers/storage.rs:612]:
```suggestion
        assert_eq!(storage.committees_iter().collect(), *committees);
```
applies to others as well

**Review by ljedrz — APPROVED** [review]

File: node/narwhal/src/helpers/storage.rs (hunk 2 above)
Comment (vvp) [reply node/narwhal/src/helpers/storage.rs:612]: Compiler complains about "type annotations needed" if I remove them :disappointed:

Comment (ljedrz) [reply node/narwhal/src/helpers/storage.rs:612]: ah, it's possible with the change from `&Vec<T>` to `&[T]`; nvm then :ok_hand:; the left-hand side might need a reference now, though

**Comment by vvp** [comment]
The CI failure on `node-cdn` was unrelated to this PR. 

As this is a test-only PR, merging now. Thank you for the review :pray:

**Pull request merged**